tools/packaging/common/istio-iptables.sh

```diff
@@ -59,31 +60,61 @@ function dump {
     ip6tables-save
 }
 
-#
-#  Function check if argument is a valid ipv4 or ipv6 address
-#
 function isValidIP() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   if isIPv4 "${1}"; then
       true
-   elif [ "$1" != "${1#*:[0-9a-fA-F]}" ]; then
+   elif isIPv6 "${1}"; then
       true
    else
       false
    fi
 }
-
 #
 # Function return true if agrument is a valid ipv4 address
 #
 function isIPv4() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   local ipv4regexp="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
```

Comment:
How about removing this ( we didn't have this validation before ), and treat it as !isPIv6 ?

Reply:
if something makes something more robust at very little cost/impact, why not to add it?

Reply:
Because it's not making things more robust - and it's not little cost. 
The pod and config will still be bad and user will have no idea about it - unlike Galley validation, the config is applied and user doesn't have any signal it happened ( nobody reads the logs from iptables script ). And adding all this complex shell for no benefit - on a critical script that too few 
people understand - is IMHO a high cost.